src/tools/generate/generate_chapters.js

```diff
@@ -22,14 +36,53 @@ const generate_chapters = async () => {
 
       const markdown = await fs.readFile(file, 'utf-8');
       const { metadata, body, toc } = await parse_file(markdown);
+      check_meta_dates(metadata);
       await write_template(language, year, chapter, metadata, body, toc);
+      await write_sitemap(language, year, chapter, metadata);
     } catch (error) {
       console.error(error);
       console.error('  Failed to generate chapter, moving onto the next one. ');
     }
   }
+  await write_sitemap_end();
```

Comment:
I know it's an awkward challenge, but instead of `write_sitemap_start`, a number of  `write_sitemap` and then `write_sitemap_end`, would it make sense to have the `write_sitemap` return the data (which we can keep in an array) and then only open and write the file once? That way we don't have to keep opening and appending to the same file.

Reply:
Yes - should have done this like this originally.

I also think it would be a much better process for the future anyway as alt languages should be handled differently than I do currently (should be `<xhtml:link ...>` links rather than separate entries as I have currently (see: https://support.google.com/webmasters/answer/189077?hl=en). This change would make doing that much easier!